analysis/analysis.go

```diff
@@ -0,0 +1,104 @@
+package analysis
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+const Doc = `This is a custom vet check for the cilium codebase.`
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "cilium",
+	Doc:      Doc,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+	Run:      run,
+}
+
+type checker interface {
+	check(n ast.Node) (bool, string, token.Pos)
+}
+
+type visitor func(ast.Node) bool
+
+func (v visitor) Visit(node ast.Node) ast.Visitor {
+	if v(node) {
+		return v
+	}
+	return nil
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	inspct, ok := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	if !ok {
+		return nil, fmt.Errorf("analyzer is not type *inspector.Inspector")
+	}
+
+	nodeFilter := []ast.Node{
+		(*ast.ForStmt)(nil),
+		(*ast.File)(nil),
+		(*ast.ImportSpec)(nil),
+	}
+	checks := []checker{
+		&timeAfterCheck{},
+	}
+	inspct.Preorder(nodeFilter, func(n ast.Node) {
+		for _, c := range checks {
+			ok, report, pos := c.check(n)
+			if !ok {
+				pass.Reportf(pos, report)
+			}
+		}
+	})
+	return nil, nil
+}
+
+type timeAfterCheck struct {
+	aliases []string
+}
+
+func (t *timeAfterCheck) check(n ast.Node) (ok bool, report string, pos token.Pos) {
```

Comment:
Maybe we can start to move these individual checks into their own files. Would be good to have structure as we add more checks.